### ISSUE TITLE
Incorrect parsing of content type application/hal+json

### ISSUE THREAD

**Issue by vire** [visionmedia/superagent#816]
Hi, we switched to superagent version 1.6.1 

we have backend-response `content-type: "application/hal+json"` according to some previous issues the [parser should be updated](https://github.com/visionmedia/superagent/releases/tag/v1.6.1) to support +json types but the npm ships superagent without the right parser.

**Comment by pornel** [comment]
Sorry, it's been changed for the browser implementation. We'll make the node version catch up in the next version,.

**Comment by pornel** [comment]
Can you show response from you backend, e.g. via `curl -i`?

**Comment by vire** [comment]
```
HTTP/1.1 200 OK
Content-type: application/hal+json
Access-control-allow-origin: *
Cache-control: no-store

{
  "_embedded" : {
    "users" : [ {
      "username" : "admin",
      "userId" : 1,
      "created" : "2015-09-11T11:00:34.204Z",
      "firstName" : "John",
      "lastName" : "Doe",
      "email" : "jdoe@somedomain.com",
      "state" : "ACTIVE",
      "_links" : {
        "self" : {
          "href" : "http://localhost:4000/users/1"
        },
        "user" : {
          "href" : "http://localhost:4000/users/1{?projection}",
          "templated" : true
        }
      }
    }]
  },
  "_links" : {
    "self" : {
      "href" : "http://localhost:4000/users"
    },
    "profile" : {
      "href" : "http://localhost:4000/profile/users"
    },
    "search" : {
      "href" : "http://localhost:4000/users/search"
    }
  },
  "page" : {
    "size" : 20,
    "totalElements" : 1,
    "totalPages" : 1,
    "number" : 0
  }
}

```
res.body is `null` res.text is okay

request.parse looks like this
```
application/json: parse()
application/x-www-form-urlencoded: 
```

so `Response.prototype.parseBody ` returns null for the body

**Comment by clemf** [comment]
I am having this same issue. res.body is null, while res.text works. Here is `curl -i` for the resource I am testing:

```sh
HTTP/1.1 200 OK
Server: nginx/1.7.6
Date: Tue, 29 Dec 2015 13:29:54 GMT
Content-Type: application/hal+json
Transfer-Encoding: chunked
Connection: keep-alive
X-Powered-By: PHP/5.6.0
Set-Cookie: PHPSESSID=o067841oqm79jvsoqjjsjdh356; path=/
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Pragma: no-cache

{"count":10,"total":10,"collectionTotal":10,"_links":{"self":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices?page=1"},"first":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices"},"last":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices?page=1"}},"_embedded":{"practice":[{"id":"0a29d33e-bcfb-4c83-b5d9-e2466b451bca","name":"Marvin-Kertzmann","status":"Enabled","contactName":"Prof. Addie Osinski Sr.","locations":{},"_links":{"self":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices\/0a29d33e-bcfb-4c83-b5d9-e2466b451bca"}}},{"id":"57c910c7-655d-467d-a72d-117e5702cf46","name":"Davis PLC","status":"Enabled","contactName":"Anita Breitenberg","locations":{},"_links":{"self":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices\/57c910c7-655d-467d-a72d-117e5702cf46"}}},{"id":"9142fb4f-2b4e-493f-977f-09b2c9b7d7e6","name":"Littel, Gerlach and Gibson","status":"Disabled","contactName":"Audrey Langworth","locations":{},"_links":{"self":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices\/9142fb4f-2b4e-493f-977f-09b2c9b7d7e6"}}},{"id":"a770faf8-e2a1-41f4-a42d-d797b1c4eaf0","name":"Labadie-Jaskolski","status":"Enabled","contactName":"Prof. Chelsey Schulist","locations":{},"_links":{"self":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices\/a770faf8-e2a1-41f4-a42d-d797b1c4eaf0"}}},{"id":"a997391c-e7a4-4e39-a22e-62ff03357ee5","name":"Reichel Ltd","status":"Disabled","contactName":"Maude Zulauf PhD","locations":{},"_links":{"self":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices\/a997391c-e7a4-4e39-a22e-62ff03357ee5"}}},{"id":"abc9062c-e036-4c5f-a4a8-8becb7b425b9","name":"Raynor, Kassulke and Mayert","status":"Disabled","contactName":"Aleen Mante Sr.","locations":{},"_links":{"self":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices\/abc9062c-e036-4c5f-a4a8-8becb7b425b9"}}},{"id":"b93ae207-12c1-4924-a438-f1f5ea66f82c","name":"Mitchell-Lebsack","status":"Disabled","contactName":"Thelma McClure","locations":{},"_links":{"self":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices\/b93ae207-12c1-4924-a438-f1f5ea66f82c"}}},{"id":"cd2b8fd5-82d2-45a8-87b3-6b09f0fd32a4","name":"Mohr and Sons","status":"Disabled","contactName":"Prof. Rex Konopelski","locations":{},"_links":{"self":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices\/cd2b8fd5-82d2-45a8-87b3-6b09f0fd32a4"}}},{"id":"d42a9eed-5835-4519-94f0-de52d3cb1860","name":"Schowalter, Waters and Nikolaus","status":"Disabled","contactName":"Rosalee Walsh","locations":{},"_links":{"self":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices\/d42a9eed-5835-4519-94f0-de52d3cb1860"}}},{"id":"de2cab91-32d8-40d1-a755-86c94783f852","name":"Hackett LLC","status":"Enabled","contactName":"Palma Windler","locations":{},"_links":{"self":{"href":"https:\/\/devlocal.audiologyedge.com\/api\/v1\/practices\/de2cab91-32d8-40d1-a755-86c94783f852"}}}]},"page_count":1,"page_size":25,"total_items":10,"page":1}
```

**Comment by niieani** [comment]
Same problem with `Content-Type:  application/x-json`.

**Comment by pornel** [comment]
`application/x-json` is not JSON, but a made-up non-standard type. Please use `application/json` or `application/something+json` only.

**Comment by niieani** [comment]
I don't control the server in this instance. 
In that case could there be a way to force `superagent` to treat the output as if it's some `Content-Type`, essentially overriding the `Content-Type` that the server returns?

**Comment by pornel** [comment]
Try 

```
superagent.get(url)
  .parse(superagent.parse['application/json'])
  .then(response => …)
```

**Comment by craigrich** [comment]
This worked for me (FF-47)
```
superagent.get(url)
  .set('Accept', 'application/hal+json')
  .then(response => …)
```

**Comment by pornel** [comment]
This has been fixed around version 1.8
